sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/tests/ConfigurationManagerTests.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System;
+using NUnit.Framework;
+
+namespace Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents.Tests
+{
+    [TestFixture]
+    public class ConfigurationManagerTests
+    {
+        private const string audienceAppId = "testAudienceAppId";
+        private const string tenantId = "testTenantId";
+        private const string authorityUrl = "oidc";
+
+        [Test]
+        public void Test_When_Attribute_Is_Null()
+        {
+            Assert.Throws<MissingFieldException>(() => new ConfigurationManager(null), "Test when attribute is null");
+        }
+
+        [Test]
+        public void Test_When_Attributes_AudienceAppId_Is_Null()
+        {
+            var attribute = new AuthenticationEventsTriggerAttribute
+            {
+                TenantId = tenantId,
+                AudienceAppId = null, // AudienceAppId is required when authority is provided.
+                AuthorityUrl = authorityUrl
+            };
+            Assert.Throws<MissingFieldException>(() => new ConfigurationManager(attribute), "Test when attribute's audienceAppId is null");
+        }
+
+        [Test]
```

Comment:
Tests from here down seems like they are similar to each other and can be combined using the TestCase tags differentiating them.

Reply:
hmm... i kind of like it this way, but if you can give me some ideas to combine them and keep them clean i am down to change it.